File: symfony-quizz/src/QuizzBundle/Entity/Game.php
<?php

namespace QuizzBundle\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * Game
 *
 * @ORM\Table(name="game")
 * @ORM\Entity(repositoryClass="QuizzBundle\Repository\GameRepository")
 */
class Game
{
    /**
     * @var int
     * @Groups({"game", "mygames", "invit"})
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /**
     * @var int
     * @Groups({"game", "mygames"})
     *
     * @ORM\Column(name="state", type="integer")
     */
    private $state;

    /**
     *
     * @Groups({"game"})
     * @ORM\ManyToOne(targetEntity="User", cascade={"persist"})
     */
    private $userA;

    /**
     * @Groups({"game"})
     *
     * @ORM\ManyToOne(targetEntity="User", cascade={"persist"})
     */
    private $userB;

    /**
     * @var User
     * @Groups({"game", "mygames"})
     *
     *
     */
    private $adv;

    /**
     * @var User
     * @Groups({"game", "mygames"})
     * @ORM\ManyToOne(targetEntity="User", cascade={"persist"})
     *
     *
     */
    private $winner;

    /**
     * @var integer
     * @Groups({"game", "mygames"})
     * @ORM\Column(name="pointsA", type="integer", nullable=true)
     *
     *
     */
    private $pointsA;

    /**
     * @var integer
     * @Groups({"game", "mygames"})
     * @ORM\Column(name="pointsB", type="integer", nullable=true)
     *
     *
     */
    private $pointsB;

    /**
     * @var \DateTime
     * @ORM\Column(name="creation_date", type="datetime", nullable=true)
     */
    private $creationDate;

    /**
     * @var \DateTime
     * @ORM\Column(name="begin_date", type="datetime", nullable=true)
     */
    private $beginDate;

    /**
     * @var \DateTime
     * @ORM\Column(name="end_date", type="datetime", nullable=true)
     */
    private $endDate;


    /**
     * Get id
     *
     * @return int
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set state
     *
     * @param integer $state
     *
     * @return Game
     */
    public function setState($state)
    {
        $this->state = $state;

        return $this;
    }

    /**
     * Get state
     *
     * @return int
     */
    public function getState()
    {
        return $this->state;
    }

    /**
     * @return User
     */
    public function getUserA()
    {
        return $this->userA;
    }

    /**
     * @param User $userA
     */
    public function setUserA(User $userA)
    {
        $this->userA = $userA;
    }

    /**
     * @return User
     */
    public function getUserB()
    {
        return $this->userB;
    }

    /**
     * @param User $userB
     */
    public function setUserB(User $userB)
    {
        $this->userB = $userB;
    }

    /**
     * @return User
     */
    public function getAdv()
    {
        return $this->adv;
    }

    /**
     * @param User $adv
     */
    public function setAdv(User $adv = null)
    {
        $this->adv = $adv;
    }

    /**
     * @return User
     */
    public function getWinner()
    {
        return $this->winner;
    }

    /**
     * @param User $winner
     */
    public function setWinner(User $winner = null)
    {
        $this->winner = $winner;
    }

    /**
     * @return int
     */
    public function getPointsA()
    {
        return $this->pointsA;
    }

    /**
     * @param int $pointsA
     */
    public function setPointsA($pointsA)
    {
        $this->pointsA = $pointsA;
    }

    /**
     * @return int
     */
    public function getPointsB()
    {
        return $this->pointsB;
    }

    /**
     * @param int $pointsB
     */
    public function setPointsB($pointsB)
    {
        $this->pointsB = $pointsB;
    }

    /**
     * @return \DateTime
     */
    public function getBeginDate()
    {
        return $this->beginDate;
    }

    /**
     * @param \DateTime $beginDate
     */
    public function setBeginDate($beginDate)
    {
        $this->beginDate = $beginDate;
    }

    /**
     * @return \DateTime
     */
    public function getEndDate()
    {
        return $this->endDate;
    }

    /**
     * @param \DateTime $endDate
     */
    public function setEndDate($endDate)
    {
        $this->endDate = $endDate;
    }

    /**
     * @return \DateTime
     */
    public function getCreationDate()
    {
        return $this->creationDate;
    }

    /**
     * @param \DateTime $creationDate
     */
    public function setCreationDate($creationDate)
    {
        $this->creationDate = $creationDate;
    }
}

File: android-quizz/app/src/main/java/com/quizz/entities/Game.java
package com.quizz.entities;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Game implements Parcelable
{

    @SerializedName("id")
    @Expose
    private Integer id;
    @SerializedName("state")
    @Expose
    private Integer state;
    @SerializedName("userA")
    @Expose
    private User userA;
    @SerializedName("userB")
    @Expose
    private User userB;
    @SerializedName("adv")
    @Expose
    private User adv;
    @SerializedName("winner")
    @Expose
    private User winner;
    @SerializedName("pointsA")
    @Expose
    private Integer pointsA;
    @SerializedName("pointsB")
    @Expose
    private Integer pointsB;
    public final static Parcelable.Creator<Game> CREATOR = new Creator<Game>() {


        @SuppressWarnings({
                "unchecked"
        })
        public Game createFromParcel(Parcel in) {
            return new Game(in);
        }

        public Game[] newArray(int size) {
            return (new Game[size]);
        }

    }
            ;

    protected Game(Parcel in) {
        this.id = ((Integer) in.readValue((Integer.class.getClassLoader())));
        this.state = ((Integer) in.readValue((Integer.class.getClassLoader())));
        this.userA = ((User) in.readValue((User.class.getClassLoader())));
        this.userB = ((User) in.readValue((User.class.getClassLoader())));
        this.adv = ((User) in.readValue((User.class.getClassLoader())));
        this.winner = ((User) in.readValue((User.class.getClassLoader())));
        this.pointsA = ((Integer) in.readValue((Integer.class.getClassLoader())));
        this.pointsB = ((Integer) in.readValue((Integer.class.getClassLoader())));
    }

    /**
     * No args constructor for use in serialization
     *
     */
    public Game() {
    }

    /**
     *
     * @param pointsB
     * @param id
     * @param pointsA
     * @param userA
     * @param userB
     * @param state
     * @param winner
     * @param adv
     */
    public Game(Integer id, Integer state, User userA, User userB, User adv, User winner, Integer pointsA, Integer pointsB) {
        super();
        this.id = id;
        this.state = state;
        this.userA = userA;
        this.userB = userB;
        this.adv = adv;
        this.winner = winner;
        this.pointsA = pointsA;
        this.pointsB = pointsB;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Game withId(Integer id) {
        this.id = id;
        return this;
    }

    public Integer getState() {
        return state;
    }

    public void setState(Integer state) {
        this.state = state;
    }

    public Game withState(Integer state) {
        this.state = state;
        return this;
    }

    public User getUserA() {
        return userA;
    }

    public void setUserA(User userA) {
        this.userA = userA;
    }

    public Game withUserA(User userA) {
        this.userA = userA;
        return this;
    }

    public User getUserB() {
        return userB;
    }

    public void setUserB(User userB) {
        this.userB = userB;
    }

    public Game withUserB(User userB) {
        this.userB = userB;
        return this;
    }

    public User getAdv() {
        return adv;
    }

    public void setAdv(User adv) {
        this.adv = adv;
    }

    public Game withAdv(User adv) {
        this.adv = adv;
        return this;
    }

    public User getWinner() {
        return winner;
    }

    public void setWinner(User winner) {
        this.winner = winner;
    }

    public Game withWinner(User winner) {
        this.winner = winner;
        return this;
    }

    public Integer getPointsA() {
        return pointsA;
    }

    public void setPointsA(Integer pointsA) {
        this.pointsA = pointsA;
    }

    public Game withPointsA(Integer pointsA) {
        this.pointsA = pointsA;
        return this;
    }

    public Integer getPointsB() {
        return pointsB;
    }

    public void setPointsB(Integer pointsB) {
        this.pointsB = pointsB;
    }

    public Game withPointsB(Integer pointsB) {
        this.pointsB = pointsB;
        return this;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(id);
        dest.writeValue(state);
        dest.writeValue(userA);
        dest.writeValue(userB);
        dest.writeValue(adv);
        dest.writeValue(winner);
        dest.writeValue(pointsA);
        dest.writeValue(pointsB);
    }

    public int describeContents() {
        return 0;
    }

}